homeassistant/components/generic/config_flow.py

```diff
@@ -515,3 +537,97 @@ async def get(self, request: web.Request, flow_id: str) -> web.Response:
+
+    async def start(self, timeout=600):
+        """Start the stream with a timeout."""
+
+        async def _timeout() -> None:
+            _LOGGER.debug("Starting preview stream with timeout %ss", timeout)
+            await asyncio.sleep(timeout)
+            _LOGGER.info("Preview stream stopping due to timeout")
+            await self.stream.stop()
+
+        await self.stream.start()
+        self._deferred_stop = self.stream.hass.async_create_task(_timeout())
+
+    async def stop(self):
+        """Stop the stream."""
+        if not self._deferred_stop.done():
+            self._deferred_stop.cancel()
+        await self.stream.stop()
+
+    def add_provider(self, provider):
+        """Add a provider to the stream."""
+        return self.stream.add_provider(provider)
+
+    def endpoint_url(self, fmt: str) -> str:
+        """Return the endpoint URL."""
+        return self.stream.endpoint_url(fmt)
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "generic_camera/start_preview",
+        vol.Required("flow_id"): str,
+        vol.Required("flow_type"): vol.Any("config_flow"),
+        vol.Required("user_input"): dict,
+    }
+)
+@websocket_api.async_response
+async def ws_start_preview(
+    hass: HomeAssistant,
+    connection: websocket_api.ActiveConnection,
+    msg: dict[str, Any],
+) -> None:
+    """Generate websocket handler for the camera still/stream preview."""
+    _LOGGER.debug("Generating websocket handler for generic camera preview")
+    ha_still_url = None
+    ha_stream_url = None
+
+    flow_id = msg["flow_id"]
+    flow = cast(
+        GenericIPCamConfigFlow,
+        hass.config_entries.flow._progress.get(flow_id),  # noqa: SLF001
+    ) or cast(
+        GenericOptionsFlowHandler,
+        hass.config_entries.options._progress.get(flow_id),  # noqa: SLF001
+    )
+    user_input = flow.preview_cam
+
+    # Create an EntityPlatform, needed for name translations
+    platform = await async_prepare_setup_platform(hass, {}, CAMERA_DOMAIN, DOMAIN)
+    entity_platform = EntityPlatform(
+        hass=hass,
+        logger=_LOGGER,
+        domain=CAMERA_DOMAIN,
+        platform_name=DOMAIN,
+        platform=platform,
+        scan_interval=timedelta(seconds=3600),
+        entity_namespace=None,
+    )
+    await entity_platform.async_load_translations()
+
+    if user_input.get(CONF_STILL_IMAGE_URL):
+        ha_still_url = f"/api/generic/preview_flow_image/{msg['flow_id']}?t={datetime.now().isoformat()}"
+        _LOGGER.debug("Got preview still URL: %s", ha_still_url)
+
+    if ha_stream := flow.preview_stream:
+        ha_stream_url = ha_stream.endpoint_url(HLS_PROVIDER)
+        _LOGGER.debug("Got preview stream URL: %s", ha_stream_url)
+
+    connection.send_result(msg["id"])
+    connection.send_message(
+        websocket_api.event_message(
+            msg["id"],
+            {"attributes": {"stillUrl": ha_still_url, "streamUrl": ha_stream_url}},
```

Comment:
Should naming style for hese be `still_url` and `stream_url`?

Reply:
Agreed.

Updated in 7ebd328

And 7e44840 on the frontend PR